<a href="https://colab.research.google.com/github/STR4NG3R/AnalisisDatos/blob/main/1U/2/FifaAnalisisExclusivo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Titulo: Casos Especiales Fifa
#Alumno: Pablo Eduardo Martinez Solis
#Fecha 02/09/2022

#Objetivo
Analisar, explorar datos, para recaudar variables de interes, los cuales nos permiten tener una mejor compresion de los jugadores segun sus estadisticas

In [2]:
import pandas as pd 
import matplotlib.pyplot as plt

Cargar datos

In [9]:
data = pd.read_csv("https://raw.githubusercontent.com/rpizarrog/Analisis-Inteligente-de-datos/main/datos/datos.FIFA.limpios.csv", encoding= 'unicode_escape')

Describir los datos a explorar los primeros 10 registros

In [10]:
data.head(10)

Unnamed: 0               Name  Age Nationality  Overall  Potential  \
0           1           L. Messi   31   Argentina       94         94   
1           2  Cristiano Ronaldo   33    Portugal       94         94   
2           3          Neymar Jr   26      Brazil       92         93   
3           4             De Gea   27       Spain       91         93   
4           5       K. De Bruyne   27     Belgium       91         92   
5           6          E. Hazard   27     Belgium       91         91   
6           7          L. Modric   32     Croatia       91         91   
7           8          L. Suárez   31     Uruguay       91         91   
8           9       Sergio Ramos   32       Spain       91         91   
9          10           J. Oblak   25    Slovenia       90         93   

                  Club Preferred.Foot  International.Reputation  Weak.Foot  \
0         FC Barcelona           Left                       5.0        4.0   
1             Juventus          Right                       5.0        4.0   
2  Paris Saint-Germain          Right                       5.0        5.0   
3    Manchester United          Right                       4.0        3.0   
4      Manchester City          Right                       4.0        5.0   
5              Chelsea          Right                       4.0        4.0   
6          Real Madrid          Right                       4.0        4.0   
7         FC Barcelona          Right                       5.0        4.0   
8          Real Madrid          Right                       4.0        3.0   
9      Atlético Madrid          Right                       3.0        3.0   

   ...  StandingTackle SlidingTackle GKDiving  GKHandling  GKKicking  \
0  ...            28.0          26.0      6.0        11.0       15.0   
1  ...            31.0          23.0      7.0        11.0       15.0   
2  ...            24.0          33.0      9.0         9.0       15.0   
3  ...            21.0          13.0     90.0        85.0       87.0   
4  ...            58.0          51.0     15.0        13.0        5.0   
5  ...            27.0          22.0     11.0        12.0        6.0   
6  ...            76.0          73.0     13.0         9.0        7.0   
7  ...            45.0          38.0     27.0        25.0       31.0   
8  ...            92.0          91.0     11.0         8.0        9.0   
9  ...            12.0          18.0     86.0        92.0       78.0   

   GKPositioning  GKReflexes      Valor  Estatura  PesoKgs  
0           14.0         8.0  110500000      1.70    72.12  
1           14.0        11.0   77000000      1.88    83.01  
2           15.0        11.0  118500000      1.75    68.04  
3           88.0        94.0   72000000      1.93    76.20  
4           10.0        13.0  102000000      1.80    69.85  
5            8.0         8.0   93000000      1.73    73.94  
6           14.0         9.0   67000000      1.73    66.22  
7           33.0        37.0   80000000      1.83    86.18  
8            7.0        11.0   51000000      1.83    82.10  
9           88.0        89.0   68000000      1.88    87.09  

[10 rows x 50 columns]

Describir estructura de datos

In [56]:
data.describe()

Unnamed: 0           Age       Overall     Potential  \
count  17955.000000  17955.000000  17955.000000  17955.000000   
mean    8978.000000     25.095349     66.228237     71.323197   
std     5183.306377      4.659778      6.921015      6.139378   
min        1.000000     16.000000     46.000000     48.000000   
25%     4489.500000     21.000000     62.000000     67.000000   
50%     8978.000000     25.000000     66.000000     71.000000   
75%    13466.500000     28.000000     71.000000     75.000000   
max    17955.000000     45.000000     94.000000     95.000000   

       International.Reputation     Weak.Foot   Skill.Moves      Crossing  \
count              17907.000000  17907.000000  17907.000000  17907.000000   
mean                   1.113699      2.947339      2.363042     49.751885   
std                    0.395193      0.660232      0.756256     18.351445   
min                    1.000000      1.000000      1.000000      5.000000   
25%                    1.000000      3.000000      2.000000     38.000000   
50%                    1.000000      3.000000      2.000000     54.000000   
75%                    1.000000      3.000000      3.000000     64.000000   
max                    5.000000      5.000000      5.000000     93.000000   

          Finishing  HeadingAccuracy  ...  StandingTackle  SlidingTackle  \
count  17907.000000     17907.000000  ...    17907.000000   17907.000000   
mean      45.586698        52.298878  ...       47.684202      45.644441   
std       19.511503        17.361599  ...       21.647299      21.271215   
min        2.000000         4.000000  ...        2.000000       3.000000   
25%       30.000000        44.000000  ...       27.000000      24.000000   
50%       49.000000        56.000000  ...       55.000000      52.000000   
75%       62.000000        64.000000  ...       66.000000      64.000000   
max       95.000000        94.000000  ...       93.000000      91.000000   

           GKDiving    GKHandling     GKKicking  GKPositioning    GKReflexes  \
count  17907.000000  17907.000000  17907.000000   17907.000000  17907.000000   
mean      16.587536     16.365276     16.203105      16.356062     16.682080   
std       17.662332     16.875398     16.467295      16.989954     17.917073   
min        1.000000      1.000000      1.000000       1.000000      1.000000   
25%        8.000000      8.000000      8.000000       8.000000      8.000000   
50%       11.000000     11.000000     11.000000      11.000000     11.000000   
75%       14.000000     14.000000     14.000000      14.000000     14.000000   
max       90.000000     92.000000     91.000000      90.000000     94.000000   

              Valor      Estatura       PesoKgs  
count  1.795500e+04  17907.000000  17907.000000  
mean   2.444530e+06      1.812325     75.279478  
std    5.626715e+06      0.067671      7.077378  
min    1.000000e+04      1.550000     49.900000  
25%    3.250000e+05      1.750000     69.850000  
50%    7.000000e+05      1.800000     74.840000  
75%    2.100000e+06      1.850000     79.830000  
max    1.185000e+08      2.060000    110.220000  

[8 rows x 44 columns]

## ¿Qué jugadores ganan más dinero, el top 5?

In [14]:
data.sort_values(by=['Valor'], ascending=False).head(5)

Unnamed: 0          Name  Age Nationality  Overall  Potential  \
2            3     Neymar Jr   26      Brazil       92         93   
0            1      L. Messi   31   Argentina       94         94   
4            5  K. De Bruyne   27     Belgium       91         92   
5            6     E. Hazard   27     Belgium       91         91   
15          16     P. Dybala   24   Argentina       89         94   

                   Club Preferred.Foot  International.Reputation  Weak.Foot  \
2   Paris Saint-Germain          Right                       5.0        5.0   
0          FC Barcelona           Left                       5.0        4.0   
4       Manchester City          Right                       4.0        5.0   
5               Chelsea          Right                       4.0        4.0   
15             Juventus           Left                       3.0        3.0   

    ...  StandingTackle SlidingTackle GKDiving  GKHandling  GKKicking  \
2   ...            24.0          33.0      9.0         9.0       15.0   
0   ...            28.0          26.0      6.0        11.0       15.0   
4   ...            58.0          51.0     15.0        13.0        5.0   
5   ...            27.0          22.0     11.0        12.0        6.0   
15  ...            20.0          20.0      5.0         4.0        4.0   

    GKPositioning  GKReflexes      Valor  Estatura  PesoKgs  
2            15.0        11.0  118500000      1.75    68.04  
0            14.0         8.0  110500000      1.70    72.12  
4            10.0        13.0  102000000      1.80    69.85  
5             8.0         8.0   93000000      1.73    73.94  
15            5.0         8.0   89000000      1.78    74.84  

[5 rows x 50 columns]

¿Cuántos jugadores de fútbol son izquierdos?

In [58]:
data.groupby(['Preferred.Foot'])['Preferred.Foot'].count()['Left']

4159

¿Cuál es el valor medio (promedio) de estaura en cms o metros de jugadores de fútbol?

In [25]:
data[['Estatura']].mean()

Estatura    1.812325
dtype: float64

¿Cómo se llamna los jugadores más bajitos en estatura?


In [30]:
data[['Name','Estatura']].sort_values(by=['Estatura']).head(10)

Name  Estatura
6474     H. Nakagawa      1.55
6172      N. Barrios      1.55
16914   K. Yamaguchi      1.55
2732           Élton      1.57
2462        J. Plata      1.57
5735      C. Techera      1.57
8392   C. Songkrasin      1.57
1329      M. Moralez      1.57
7004       A. Araujo      1.60
5531      E. Oztumer      1.60

De qué nacionalidad o país hay mas jugadres registrados?


In [55]:
data.groupby('Nationality')['Nationality'].count().sort_values(ascending=False).head(1)

Nationality
England    1660
Name: Nationality, dtype: int64

#Interpretacion
Podemos observar que gracias a la coleccion de datos podemos apreciar que la mayoria de jugadores vienen de England, se puede apreciar que la estatura promedio de los jugadores es de 1.81, las edades promedios son de 25, hay mas jugadores derechos que zurdos